src/lbaf/IO/configValidator.py

```diff
@@ -0,0 +1,123 @@
+from logging import Logger
+
+import yaml
+from schema import And, Optional, Or, Schema, Use
+
+from lbaf.Utils.logger import logger
+
+
+class ConfigValidator:
+    """ Validates schema of VT Object Map files (json)
+    """
+    def __init__(self, config_to_validate: dict, logger: Logger):
+        self.__config_to_validate = config_to_validate
+        self.__skeleton = Schema({
+            Or("from_data", "from_samplers", only_one=True): dict,
+            "work_model": {"name": And(str, lambda c: c in ("LoadOnly", "AffineCombination"),
+                                       error="LoadOnly or AffineCombination needs to be chosen!"),
+                           "parameters": {"alpha": float,
+                                          "beta": float,
+                                          "gamma": float}},
+            "algorithm": {
+                "name": And(str, lambda d: d in ("InformAndTransfer", "BruteForce"),
+                            error="InformAndTransfer or BruteForce needs to be chosen!"),
+                Optional("parameters"): dict
+            },
+            "exodus": bool,
+            "x_procs": int,
+            "y_procs": int,
+            "z_procs": int,
+            "output_file_stem": str,
+            Optional("brute_force_optimization"): bool,
+            Optional("logging_level"): And(str, Use(str.lower), lambda f: f in ("info", "debug", "warning", "error"),
+                                           error="info or debug or warning or error needs to be chosen!"),
+            Optional("terminal_background"): And(str, Use(str.lower), lambda g: g in ("light", "dark"),
+                                                 error="light or dark needs to be chosen!"),
+            Optional("output_dir"): str,
+            Optional("generate_multimedia"): bool
+        })
+        self.__from_data = Schema({"data_stem": str, "phase_id": int})
+        self.__from_samplers = Schema({
+            "n_objects": int,
+            "n_mapped_ranks": int,
+            "communication_degree": int,
+            "time_sampler": {"name": And(str, Use(str.lower), lambda a: a in ("uniform", "lognormal"),
+                                         error="uniform or lognormal needs to be chosen!"),
+                             "parameters": [float, float]},
+            "volume_sampler": {"name": And(str, Use(str.lower), lambda b: b in ("uniform", "lognormal"),
+                                           error="uniform or lognormal needs to be chosen!"),
```

Comment:
Same here: we should not hard-code the list of available options and at the same time the error message: this leads to database de-normalization as it becomes possible to have an error message not up-to-date with the list of available functions. This list must be defined only **once** and used by both the `lambda` and the `error`.

Reply:
I will put the allowed values in `configurationValidator.py`, we could put them into other place if it exists.